src/main/scala/rules/Brancher.scala

```diff
@@ -87,40 +95,33 @@ object brancher extends BranchingRules {
          * needed, the second one ensures that the current path conditions (etc.) of the
          * "offloading" verifier are captured.
          */
-//        val functionsOfCurrentDecider = v.decider.freshFunctions
-//        val macrosOfCurrentDecider = v.decider.freshMacros
-//        val pcsOfCurrentDecider = v.decider.pcs.duplicate()
-
-//        println(s"\n[INIT elseBranchVerificationTask v.uniqueId = ${v.uniqueId}]")
-//        println(s"  condition = $condition")
-//        println("  v.decider.pcs.assumptions = ")
-//        v.decider.pcs.assumptions foreach (a => println(s"    $a"))
-//        println("  v.decider.pcs.branchConditions = ")
-//        v.decider.pcs.branchConditions foreach (a => println(s"    $a"))
+        if (parallelizeElseBranch){
+          functionsOfCurrentDecider = v.decider.freshFunctions
+          macrosOfCurrentDecider = v.decider.freshMacros
+          pcsOfCurrentDecider = v.decider.pcs.duplicate()
+        }
 
         (v0: Verifier) => {
           SymbExLogger.currentLog().switchToNextBranch(uidBranchPoint)
           SymbExLogger.currentLog().markReachable(uidBranchPoint)
-          executionFlowController.locally(s, v0)((s1, v1) => {
-            if (v.uniqueId != v1.uniqueId) {
+          if (v.uniqueId != v0.uniqueId){
+            /* [BRANCH-PARALLELISATION] */
+            // executing the else branch on a different verifier, need to adapt the state
 
-              /* [BRANCH-PARALLELISATION] */
-              throw new RuntimeException("Branch parallelisation is expected to be deactivated for now")
+            val newFunctions = functionsOfCurrentDecider -- v0.decider.freshFunctions
```

Comment:
If I remember correctly, the operands are both `ListSet`s, i.e. linked lists, with list subtraction in `O(n^2)`. Not your doing, of course, but maybe you could a comment that points this out.

Side remark: I wanted to replace these sets by more performant ones since forever, but couldn't find a great alternative. Java's `LinkedHashSet` might be a good candidate, but it's not a native Scala collection ...